cmake/modules/CheckSubmodule.cmake

```diff
@@ -0,0 +1,67 @@
+# Utility for validating and -- if needed -- cloning a git submodule
+# This project depends on "CheckCommand", which is part of the LMMS project.
+#
+# Looks for a relative file inside source directory of current target
+# - Attempts to clone with git if it's missing
+# - Throws FATAL_ERROR by default unless OPTIONAL is specified
+# - If submodule "foo" is located inside "src/3rdparty/foo"
+#   - RELATIVE_FILE: foo/bar.c
+#   - resolving to:  /src/3rdparty/foo/foo/bar.c
+#
+# Usage:
+#       CHECK_SUBMODULE(foo/bar.c)
+#       CHECK_SUBMODULE(foo/bar.c OPTIONAL)
+#
+# Copyright (c) 2017, Tres Finocchiaro, <tres.finocchiaro@gmail.com>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+INCLUDE(CheckCommand)
+
+MACRO(CHECK_SUBMODULE)
+	SET(ARG_LIST "${ARGN}")
+	LIST(LENGTH ARG_LIST ARG_LEN)
+	IF (ARG_LEN EQUAL 1)
+		LIST(GET ARG_LIST 0 RELATIVE_FILE)
+	ELSEIF(ARG_LEN EQUAL 2)
+		LIST(GET ARG_LIST 0 RELATIVE_FILE)
+		LIST(GET ARG_LIST 1 OPTIONAL)
+	ELSE()
+		MESSAGE(FATAL_ERROR "CHECK_SUBMODULE takes (1) or (2) arguments but (${ARG_LEN}) were provided")
+	ENDIF()
+	
+	# Guess submodule name from the path prefix
+	STRING(REPLACE "/"  ";" PARTS ${RELATIVE_FILE})
+	LIST(GET PARTS 0 SUBMODULE_NAME)
+	SET(ABSOLUTE_FILE "${CMAKE_CURRENT_SOURCE_DIR}/${RELATIVE_FILE}")
+	SET(SUBMODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${SUBMODULE_NAME}")
+
+	# Attempt to do lazy clone
+	IF(NOT EXISTS "${ABSOLUTE_FILE}")
+		MESSAGE("-- Submodule '${SUBMODULE_NAME}' missing")
+		MESSAGE("--   Cloning '${SUBMODULE_NAME}' using git")
+		CHECK_COMMAND(git)
```

Comment:
I think it's better to just use `FIND_PACKAGE(Git)` here instead. No need for an extra CMake module.

Reply:
I'll switch to `FIND_PACKAGE(Git)` and use `GIT_EXECUTABLE` variable, thanks.  This module was originally added to look for some of the automake/autoconf/autopoint tools that weren't trivial to locate, but I've since realized we don't need them if we build with cmake.  Thanks.  I'll retire the module for now.